source/MRMesh/MRMarchingCubes.cpp

```diff
@@ -502,6 +386,9 @@ Expected<TriMesh> volumeToMesh( const V& volume, const MarchingCubesParams& para
         const auto layerEnd = std::min( ( blockIndex + 1 ) * layerPerBlockCount, layerCount );
 
         const VoxelsVolumeAccessor<V> acc( volume );
+        /// grid point with integer coordinates (0,0,0) will be shifted to this position in 3D space
+        const Vector3f zeroPoint = params.origin + mult( acc.shift(), volume.voxelSize );
```

Comment:
does it produce half voxel difference between VdbVolume and SimpleVolume?

Reply:
no, this half-voxel is inside `acc.shift()`